### PR TITLE
TINY-8441: Updated docs links for 6.0

### PR DESCRIPTION
**Pull request by EkimChau**
Related Ticket:

Description of Changes:
* Updated embedded URL links to docs from within the core source code
* Updated help dialog plugin links to the docs
* Updated docs repo to fill missing landing pages

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/Deprecations.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ const logRemovedWarnings = (rawOptions: RawEditorOptions, normalizedOptions: Nor
     // eslint-disable-next-line no-console
     console.warn(
       'The following deprecated features are currently enabled and have been removed in TinyMCE 6.0. These features will no longer work and should be removed from the TinyMCE configuration. ' +
-      'See https://www.tiny.cloud/docs/release-notes/6.0-upcoming-changes/ for more information.' +
+      'See https://www.tiny.cloud/docs/tinymce/6/6_0_0-release-notes-overview.html for more information.' +
```
Comment (TheSpyder) [inline modules/tinymce/src/core/main/ts/Deprecations.ts:51]: This should probably be a migration guide, not the release notes, but I guess the release notes will include a link to the guide 🤔

File: modules/tinymce/src/plugins/quickbars/demo/html/demo.html [hunk 2]
```diff
@@ -36,8 +36,8 @@ <h2>The world’s first rich text editor in the cloud</h2>
 
 
         <p>
-          One of these enhancements is <strong>Tiny Drive</strong>: imagine file management for TinyMCE, in the cloud, made super easy.
-          Learn more at <a href='https://www.tiny.cloud/tinydrive/'>tiny.cloud/tinydrive</a>, where you’ll find a working demo and an opportunity to provide feedback to the product team.
+          One of these enhancements is <strong>Drive</strong>: imagine file management for TinyMCE, in the cloud, made super easy.
```
Comment (TheSpyder) [inline modules/tinymce/src/plugins/quickbars/demo/html/demo.html:39]: isn't the product name still `Tiny Drive`?

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/plugins/quickbars/demo/html/demo.html (hunk 2 above)
Comment (EkimChau) [reply modules/tinymce/src/plugins/quickbars/demo/html/demo.html:39]: no thats since been changed https://www.tiny.cloud/drive/

Comment (lnewson) [reply modules/tinymce/src/plugins/quickbars/demo/html/demo.html:39]: That page uses both so I don't think it's a good source of truth:
image

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -96,6 +96,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The default value for the `schema` option has been changed to `html5` #TINY-8261
 - The default value for the `table_style_by_css` option has been changed to `true` #TINY-8259
 - The default value for the `table_use_colgroups` option has been changed to `true` #TINY-8259
+- The URL links to the TinyMCE docs reference site #TINY-8441
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:99]:
```suggestion
```
This shouldn't be in the TinyMCE changelog because it's not something that's shipped as part of TinyMCE, instead it's part of the docs.

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -1396,8 +1397,6 @@ Full documentation for the version 5 features and changes is available at https:
 
 Developer preview 1
 
-Initial list of features and changes is available at https://tiny.cloud/docs-preview/release-notes/new-features/
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:1399]: This shouldn't have been removed, as now there's nothing listed there.

- [ ] Revert this change

File: modules/tinymce/src/core/main/ts/Deprecations.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/Deprecations.ts:51]: Agreed, this should link to the migration guide instead.

- [ ] Use the migration guide link instead

File: modules/tinymce/src/core/main/ts/api/EditorManager.ts [hunk 5]
```diff
@@ -360,7 +360,7 @@ const EditorManager: EditorManager = {
       if (Env.browser.isIE() || Env.browser.isEdge()) {
         ErrorReporter.initError(
           'TinyMCE does not support the browser you are using. For a list of supported' +
-          ' browsers please see: https://www.tinymce.com/docs/get-started/system-requirements/'
+          ' browsers please see: https://www.tiny.cloud/docs/tinymce/6/support.html#_supported_web_browsers'
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/EditorManager.ts:363]: Another ID that should be fixed (i.e. not using an underscore prefixed ID)

- [ ] Fix this here and in the docs to use a fixed ID

File: modules/tinymce/src/core/main/ts/api/WindowManager.ts [hunk 6]
```diff
@@ -142,15 +142,15 @@ const WindowManager = (editor: Editor): WindowManager => {
      * Opens a new window.
      *
      * @method open
-     * @param {Object} args For a list of options, see: <a href="https://www.tiny.cloud/docs/ui-components/dialog/#configurationoptions">Dialog - Configuration options</a>.
+     * @param {Object} args For a list of options, see: <a href="https://www.tiny.cloud/docs/tinymce/6/dialog-configuration.html#_configuration_options">Dialog - Configuration options</a>.
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/WindowManager.ts:145]: Another ID that should be fixed (i.e. not using an underscore prefixed ID)

- [ ] Fix this here and in the docs to use a fixed ID like it was previously

File: modules/tinymce/src/core/main/ts/api/WindowManager.ts [hunk 7]
```diff
@@ -142,15 +142,15 @@ const WindowManager = (editor: Editor): WindowManager => {
...
      */
     open,
 
     /**
      * Opens a new window for the specified url.
      *
      * @method openUrl
-     * @param {Object} args For a list of options, see: <a href="https://www.tiny.cloud/docs/ui-components/urldialog/#urldialogconfiguration">URL dialog configuration</a>.
+     * @param {Object} args For a list of options, see: <a href="https://www.tiny.cloud/docs/tinymce/6/urldialog.html#_configuration">URL dialog configuration</a>.
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/WindowManager.ts:153]: Another ID that should be fixed (i.e. not using an underscore prefixed ID)

- [ ] Fix this here and in the docs to use a fixed ID like it was previously

File: modules/tinymce/src/core/main/ts/api/fmt/StyleFormat.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { BlockFormat, InlineFormat, SelectorFormat } from './Format';
 
-// somewhat documented at https://www.tiny.cloud/docs/configure/content-formatting/#style_formats
+// somewhat documented at https://www.tiny.cloud/docs/tinymce/6/content-formatting.html#_formats
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/fmt/StyleFormat.ts:3]: Another ID that should be fixed (i.e. not using an underscore prefixed ID). This one probably doesn't matter though given it's only a comment and not something used in the generated docs.

**Review by lnewson — CHANGES_REQUESTED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 4 above)
Comment (EkimChau) [reply modules/tinymce/CHANGELOG.md:1400]: its a 404 dead link

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:1400]: Removing it is still not the right solution then... we need to fix the link or add something back so that we don't have a version without any entries. I'd suggest linking to https://www.tiny.cloud/docs/release-notes/release-notes50/#newfeatures instead.

File: SECURITY.md [hunk 9]
```diff
@@ -7,10 +7,11 @@ Tiny Technologies, Inc. supports the following community versions of TinyMCE:
 | Version | Supported                      |
 | ------- | ------------------------------ |
 | 5.10.x  | &#10004;                       |
+| 6.0.x     | &#10004;                       |
```
Comment (lnewson) [inline SECURITY.md:10]:
```suggestion
| 6.0.x    | &#10004;                       |
```
I gave a suggestion that fixed this properly, but lets try again.

File: README.md [hunk 10]
```diff
@@ -38,21 +38,21 @@ TinyMCE is easily integrated into your projects with the help of components such
 - [tinymce-vue](https://github.com/tinymce/tinymce-vue)
 - [tinymce-angular](https://github.com/tinymce/tinymce-angular)
 
-With over 29 integrations, and 400+ APIs, see the TinyMCE docs for a full list of editor [integrations](https://www.tiny.cloud/docs/integrations/).
+With over 29 integrations, and 400+ APIs, see the TinyMCE docs for a full list of editor [integrations](https://www.tiny.cloud/docs/tinymce/6/integrations.html).
```
Comment (lnewson) [inline README.md:41]: This URL worries me, it's to a WIP page and I suspect it probably needs to link to the `How-to guides` -> `Environment setup guides` as we no longer have an integrations page and that is currently the nav that has the details about the integrations. I guess worse case scenario we can redirect this single URL later.

File: modules/tinymce/CHANGELOG.md [hunk 11]
```diff
@@ -1224,7 +1224,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## 5.0.0 - 2019-02-04
 
-Full documentation for the version 5 features and changes is available at https://www.tiny.cloud/docs/release-notes/
+Full documentation for the version 5 features and changes is available at https://www.tiny.cloud/docs/tinymce/5/5_0_0-release-notes-overview.html
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:1227]:
```suggestion
Full documentation for the version 5 features and changes is available at https://www.tiny.cloud/docs/tinymce/5/release-notes/release-notes50.html
```
This link 404s/doesn't exist and I believe this suggestion is the correct one.

File: modules/tinymce/CHANGELOG.md [hunk 12]
```diff
@@ -1397,7 +1397,7 @@ Full documentation for the version 5 features and changes is available at https:
 
 Developer preview 1
 
-Initial list of features and changes is available at https://tiny.cloud/docs-preview/release-notes/new-features/
+Initial list of features and changes is available at https://tiny.cloud/tinymce/5/release-notes/release-notes50.html
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:1400]:
```suggestion
Initial list of features and changes is available at https://www.tiny.cloud/docs/tinymce/5/release-notes/release-notes50.html
```
This link is also incorrect (missing the `docs` part of the path).

**Review by lnewson — APPROVED** [review]

File: modules/tinymce/src/plugins/quickbars/demo/html/demo.html (hunk 2 above)
Comment (metricjs) [reply modules/tinymce/src/plugins/quickbars/demo/html/demo.html:39]: Realised this is a demo page, so wording isn't critical. Therefore, resolving.

File: README.md (hunk 10 above)
Comment (metricjs) [reply README.md:41]: Talked with Lee, we can deal with this with redirects if needed.

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 11 above)
Comment (metricjs) [reply modules/tinymce/CHANGELOG.md:1227]: Lee walked me through this, so accepting on Mike's behalf due to time constraints.

File: modules/tinymce/CHANGELOG.md (hunk 12 above)
Comment (metricjs) [reply modules/tinymce/CHANGELOG.md:1400]: Lee walked me through this, so accepting on Mike's behalf due to time constraints.
